scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -478,38 +478,57 @@
   "Takes an async channel that will have tx report queue elements on it"
   [tx-report-chan conn driver-ref]
   (log/info "Starting tx-report-queue")
-  (let [kill-chan (async/chan)]
+  (let [kill-chan (async/chan)
+        query-db (d/db conn)
+        query-basis (d/basis-t query-db)
+        tasks-to-kill (q '[:find ?task-id
+                           :in $ [?status ...]
+                           :where
+                           [?i :instance/status ?status]
+                           [?job :job/instance ?i]
+                           [?job :job/state :job.state/completed]
+                           [?i :instance/task-id ?task-id]]
+                         query-db [:instance.status/unknown :instance.status/running])]
+    (doseq [[task-id] tasks-to-kill]
+      (if-let [driver @driver-ref]
+        (try
+          (log/info "Attempting to kill task" task-id "from already completed job")
```

Comment:
Would it be useful to log the job uuid as well?

Reply:
I think we could log it, but not sure if the cost of querying for it is worth it since it's easy when debugging to go from one to the other, and it's not going to change.

Reply:
Makes sense